pkg/build/strategies/dockerfile/dockerfile.go

```diff
@@ -154,41 +153,31 @@ func (builder *Dockerfile) CreateDockerfile(config *api.Config) error {
 			buffer.WriteString(fmt.Sprintf("# Save-artifacts script sourced from builder image based on user input or image metadata.\n"))
 			artifactsScript = sanitize(filepath.ToSlash(filepath.Join(imageScriptsDir, "save-artifacts")))
 		}
+		// Switch to the image user if it is not root
+		if len(imageUser) > 0 && imageUser != "root" {
+			buffer.WriteString(fmt.Sprintf("USER %s\n", imageUser))
+		}
 		buffer.WriteString(fmt.Sprintf("RUN if [ -s %[1]s ]; then %[1]s > %[2]s; else touch %[2]s; fi\n", artifactsScript, artifactsTar))
 	}
 
 	buffer.WriteString(fmt.Sprintf("FROM %s\n", config.BuilderImage))
+	// Run as root to COPY and chown source content
+	buffer.WriteString("USER root\n")
```

Comment:
can we move this closer to the copy?  right now it looks like it'll come out before the label+env commands.